pkgs/development/php-packages/grpc/default.nix

```diff
@@ -1,18 +1,40 @@
-{ buildPecl
-, zlib
+{ stdenv
+, fetchFromGitHub
+, autoreconfHook
+, php
 , lib
+, grpc
 }:
 
-buildPecl {
+stdenv.mkDerivation rec {
   pname = "grpc";
-  version = "1.56.0";
+  extensionName = "grpc";
+  version = grpc.version;
 
-  sha256 = "sha256-uzxYMUzExMBDtwv3FipOuuUHg0v1wqAUtn69jXAQnf4=";
+  src = grpc.src;
 
+  sourceRoot = "source/src/php/ext/grpc";
+
+  patches = [
+    ./use-system-grpc.patch # https://github.com/grpc/grpc/pull/35404
+    ./skip-darwin-test.patch # https://github.com/grpc/grpc/pull/35403
+  ];
+
+  nativeBuildInputs = [ autoreconfHook grpc ];
+  buildInputs = [ php.unwrapped ];
+
+  autoreconfPhase = ''
+    phpize
+  '';
+
+  makeFlags = [ "EXTENSION_DIR=$(out)/lib/php/extensions" ];
+  checkPhase = "NO_INTERACTON=yes make test";
   doCheck = true;
   checkTarget = "test";
 
-  nativeBuildInputs = [ zlib ];
+  configureFlags = [
+    "--with-grpc-dir=${grpc}"
```

Comment:
Ideally, we would use pkg-config like other extensions since PHP 7.4/8. That should make the patch more upstreamable.

Reply:
I am not sure why they currently pecl zip have extension + grpc + third party libraries included and compiles everything together. I think they do that to have the same version of grpc library to php extension version, thats why I reuse the source of grpc.

I asked for pkg-config in my PR: https://github.com/grpc/grpc/pull/35404#issuecomment-1871022142

Reply:
I updated the patch to use pkg-config